pkg/kubelet/deviceplugin/utils.go

```diff
@@ -21,41 +21,16 @@ import (
 
 	"k8s.io/api/core/v1"
 	v1helper "k8s.io/kubernetes/pkg/api/v1/helper"
-	pluginapi "k8s.io/kubernetes/pkg/kubelet/apis/deviceplugin/v1alpha1"
 )
 
-func cloneDevice(d *pluginapi.Device) *pluginapi.Device {
-	return &pluginapi.Device{
-		ID:     d.ID,
-		Health: d.Health,
-	}
-
-}
-
-func copyDevices(devs map[string]*pluginapi.Device) []*pluginapi.Device {
-	var clones []*pluginapi.Device
-
-	for _, d := range devs {
-		clones = append(clones, cloneDevice(d))
-	}
-
-	return clones
-}
-
 // IsResourceNameValid returns an error if the resource is invalid or is not an
 // extended resource name.
 func IsResourceNameValid(resourceName string) error {
```

Comment:
I think this function is now only used in manager.go. Can we move it there and make it private? We can also move its test to manager_test.go, and then remove this file and its test file.

Reply:
Well, in fact, I meant to make it private at first.  Renaud stand that it is designed for potential usage from users outside and better to keep it
So, @RenaudWasTaken What's your opinion?

Reply:
My opinion is still the same :)
Depending on a method that we defined as the ResourceNameValidator provides for more maintainable code.

A simple example would be that if in the future we want to expand what we consider as a valid resource name without changing the definition of extended resources this is the only method that we will have to change.
Note that this method is also very useful for any device plugin vendor for the same reasons.

> We can also move its test to manager_test.go, and then remove this file and its test file.
 
Why are we so bent on removing this file ?
IMHO it makes little sense to have utils functions in the manager. The tests emphasize this, as the file name indicates they test utils functions.
This makes the whole module easier to understand without having to read unnecessary functions in `manager.go`.

Reply:
Right now, this function is only used in manager.go during registration. Quite likely it will stay this way because registration logic should be kept internally in device plugin. Vendors who want to implement device plugin should only include api package instead of this package. I would suggest we don't over-engineer at this time, and turn it into a private function in manager.go.